core/modules/system/system.admin.inc

```diff
@@ -282,6 +296,7 @@ function system_themes_admin_form_submit($form, &$form_state) {
   }
 
   $config->set('admin_theme', $new_admin_theme);
+  $config->set('login_admin_theme', $form_state['values']['login_admin_theme']);
```

Comment:
Only setting the config item's not sufficient here.

It's probably necessary to flush all caches, as other (cached) hooks are relevant for the simplified display.